Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py

```diff
@@ -226,6 +226,33 @@ def clear_trailing_whitespace(res):
     return res
 
 
+def filter_and_save_unseen_incident(incidents: List, limit: int, number_of_already_filtered_incidents: int) -> List:
+    """
+    Filters incidents that were seen already and saves the unseen incidents to LastRun object.
+    :param incidents: List of incident - must be list
+    :param limit: the maximum number of incident per fetch
+    :param number_of_already_filtered_incidents: number of incidents that were fetched already
+    :return: the filtered incidents.
+    """
+    last_run_obj = demisto.getLastRun()
+    fetched_starred_incidents = last_run_obj.pop('fetched_starred_incidents', {})
+    filtered_incidents = []
+    for incident in incidents:
+        incident_id = incident.get('incident_id')
+        if incident_id in fetched_starred_incidents:
+            demisto.debug(f'incident (ID {incident_id}) was already fetched in the past.')
+            continue
+        fetched_starred_incidents[incident_id] = True
```

Comment:
why do you need a dict?

Reply:
setLastRun did not work with a set as far as I remember.